src/util/logging.h

```diff
@@ -1,9 +1,91 @@
 #pragma once
 
 #include <QDir>
+#include <QLoggingCategory>
+
+/// Enable/disable verbose debug logging globally.
+///
+/// Verbose debug logging is only needed temporarily during
+/// development and probably cause performance degressions.
+/// IT SHOULD BE KEPT DISABLED IN REGULAR BUILDS!
+///
+/// After switching the default log level from Debug to Info
+/// this #define might be enabled permanently and could then
+/// finally be removed from the code.
+///
+/// Example for verbose debug logs: SoundSourceFFmpeg
+#define VERBOSE_DEBUG_LOG false
+
+/// Macro for composing parent/child logging category names
+///
+/// Use this macro to compose a new child logging category from a
+/// parent category.
+#define MIXXX_LOGGING_CATEGORY_PARENT_CHILD(parentName, childName) \
+    parentName "." childName
+
+/// Root logging category for Mixxx
+///
+/// Only use this as a parent if none of the predefined base logging
+/// categories (see below) are suitable. In this case discuss with the
+/// team if a new base category needs to be introduced.
+Q_DECLARE_LOGGING_CATEGORY(mixxxLog)
+#define MIXXX_LOGGING_CATEGORY_ROOT "mixxx"
+
+/// Base logging category for configuration
+Q_DECLARE_LOGGING_CATEGORY(mixxxLogConfig)
+#define MIXXX_LOGGING_CATEGORY_CONFIG \
+    MIXXX_LOGGING_CATEGORY_PARENT_CHILD(MIXXX_LOGGING_CATEGORY_ROOT, "config")
+
+/// Base logging category for controls and controllers
+Q_DECLARE_LOGGING_CATEGORY(mixxxLogControl)
+#define MIXXX_LOGGING_CATEGORY_CONTROL \
+    MIXXX_LOGGING_CATEGORY_PARENT_CHILD(MIXXX_LOGGING_CATEGORY_ROOT, "control")
+
+/// Base logging category for generic database access
+Q_DECLARE_LOGGING_CATEGORY(mixxxLogDatabase)
+#define MIXXX_LOGGING_CATEGORY_DATABASE \
+    MIXXX_LOGGING_CATEGORY_PARENT_CHILD(MIXXX_LOGGING_CATEGORY_ROOT, "database")
+
+/// Base logging category for hardware I/O sound devices
+Q_DECLARE_LOGGING_CATEGORY(mixxxLogDevice)
+#define MIXXX_LOGGING_CATEGORY_DEVICE \
+    MIXXX_LOGGING_CATEGORY_PARENT_CHILD(MIXXX_LOGGING_CATEGORY_ROOT, "device")
```

Comment:
```suggestion
/// Base logging category for hardware I/O sound devices
Q_DECLARE_LOGGING_CATEGORY(mixxxLogAudioDevice)
#define MIXXX_LOGGING_CATEGORY_DEVICE \
    MIXXX_LOGGING_CATEGORY_PARENT_CHILD(MIXXX_LOGGING_CATEGORY_ROOT, "audiodevice")
```

"Device" by itself could mean lots of things.